src/runtime.js

```diff
@@ -507,13 +506,15 @@ function prepareAndRegisterServiceForDocShadowMode(global, extensions,
  * @param {function(!./service/ampdoc-impl.AmpDoc):!Object=} opt_factory
  */
 function registerServiceForDoc(ampdoc, name, opt_ctor, opt_factory) {
-  dev().assert((opt_ctor || opt_factory) && (!opt_ctor || !opt_factory),
+  dev().assert(!opt_ctor != !opt_factory,
```

Comment:
I already have a check in service.js for this. Do you think this check should be deleted from here?